crates/uv/src/cli.rs

```diff
@@ -284,17 +284,26 @@ pub(crate) struct PipCompileArgs {
     /// By default, `uv` strips extras, as any packages pulled in by the extras are already included
     /// as dependencies in the output file directly. Further, output files generated with
     /// `--no-strip-extras` cannot be used as constraints files in `install` and `sync` invocations.
-    #[clap(long)]
+    #[clap(long, overrides_with("strip_extras"))]
     pub(crate) no_strip_extras: bool,
 
+    #[arg(long, overrides_with("no_strip_extras"), hide = true)]
+    pub(crate) strip_extras: bool,
+
     /// Exclude comment annotations indicating the source of each package.
-    #[clap(long)]
+    #[clap(long, overrides_with("annotate"))]
     pub(crate) no_annotate: bool,
 
+    #[arg(long, overrides_with("no_annotate"), hide = true)]
+    pub(crate) annotate: bool,
+
     /// Exclude the comment header at the top of the generated output file.
-    #[clap(long)]
+    #[clap(long, overrides_with("header"))]
     pub(crate) no_header: bool,
 
+    #[arg(long, overrides_with("no_header"), hide = true)]
+    pub(crate) header: bool,
```

Comment:
These ones are a bit weird, because the hidden value is `--header` and the visible value (and value in the configuration file) is `--no-header` / `no-header = true`. Like, in theory it should be... `--no-no-header`?

Reply:
I think it's okay as-is.